csharp/autobuilder/Semmle.Autobuild/Autobuilder.cs

```diff
@@ -113,46 +119,75 @@ public Autobuilder(IBuildActions actions, AutobuildOptions options)
             Actions = actions;
             Options = options;
 
-            pathsLazy = new Lazy<IEnumerable<string>>(() =>
+            pathsLazy = new Lazy<IEnumerable<(string, int)>>(() =>
             {
-                var files = new List<string>();
-                FindFiles(options.RootDirectory, options.SearchDepth, files);
-                return files.
-                    OrderBy(s => s.Count(c => c == Path.DirectorySeparatorChar)).
-                    ThenBy(s => Path.GetFileName(s).Length).
-                    ToArray();
+                var files = new List<(string, int)>();
+                FindFiles(options.RootDirectory, 0, options.SearchDepth, files);
+                return files.OrderBy(f => f.Item2).ToArray();
             });
 
-            solutionsToBuildLazy = new Lazy<IList<ISolution>>(() =>
+            projectsOrSolutionsToBuildLazy = new Lazy<IList<IProjectOrSolution>>(() =>
             {
                 if (options.Solution.Any())
                 {
-                    var ret = new List<ISolution>();
+                    var ret = new List<IProjectOrSolution>();
                     foreach (var solution in options.Solution)
                     {
                         if (actions.FileExists(solution))
                             ret.Add(new Solution(this, solution));
                         else
-                            Log(Severity.Error, "The specified solution file {0} was not found", solution);
+                            Log(Severity.Error, $"The specified solution file {solution} was not found");
                     }
                     return ret;
                 }
 
-                var solutions = GetExtensions(".sln").
-                        Select(s => new Solution(this, s)).
-                        Where(s => s.ProjectCount > 0).
-                        OrderByDescending(s => s.ProjectCount).
-                        ThenBy(s => s.Path.Length).
+                bool FindFiles(string extension, Func<string, ProjectOrSolution> create, out IEnumerable<IProjectOrSolution> files)
```

Comment:
I don't think this particularly needs to be a local function.

Reply:
I think local functions are nice (as opposed to private static methods) when they are only used inside one particular callable, as in this case.